### PR TITLE
[Bug](partition_topn) fix partition_topn not reset output rows after do_partition_topn_sort

### PR DESCRIPTION
**Pull request by zhangstar333**
## Proposed changes

should reset all variables about output rows after do_partition_topn_sort and get_next from sorter
if we not reset all variables. it's will be regard as  have output many rows
but the sorter still have remaing many data need to output.

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: be/src/pipeline/exec/partition_sort_sink_operator.h [hunk 1]
```diff
@@ -66,7 +66,12 @@ struct PartitionSortInfo {
 };
 
 static constexpr size_t INITIAL_BUFFERED_BLOCK_BYTES = 64 << 20;
+
+#ifndef NDEBUG
+static constexpr size_t PARTITION_SORT_ROWS_THRESHOLD = 10;
```
Comment (HappenLee) [inline be/src/pipeline/exec/partition_sort_sink_operator.h:71]: why change the default value？

Comment (HappenLee) [inline be/src/pipeline/exec/partition_sort_sink_operator.h:71]: why change the default value？

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

File: be/src/pipeline/exec/partition_sort_sink_operator.h (hunk 1 above)
Comment (zhangstar333) [reply be/src/pipeline/exec/partition_sort_sink_operator.h:71]: for debug

**Pull request merged**